src/lib.rs

```diff
@@ -766,14 +771,78 @@ pub struct PreparedQueuePair<'res> {
     rnr_retry: u8,
 }
 
+/// A Global identifier for ibv.
+///
+/// This struct acts as a rust wrapper for `ffi::ibv_gid`. We use it instead of
+/// `ffi::ibv_giv` because `ffi::ibv_gid` is actually an untagged union.
+///
+/// ```c
+/// union ibv_gid {
+///     uint8_t   raw[16];
+///     struct {
+/// 	    __be64	subnet_prefix;
+/// 	    __be64	interface_id;
+///     } global;
+/// };
+/// ```
+///
+/// It appears that `global` exists for convenience, but can be safely ignored.
+/// For continuity, the methods `subnet_prefix` and `interface_id` are provided.
+/// These methods read the array as big endian, regardless of native cpu
+/// endianness.
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Default, Copy, Clone, Debug, Eq, PartialEq)]
+struct Gid {
```

Comment:
This should probably be `#[repr(C)]`. Or maybe `#[repr(transparent)]`. I think either will do.

Reply:
I think `#[repr(transparent)]` makes more sense, but I agree that the memory layout will be identical.